### ISSUE TITLE
error encountered when supplying token in L.esri.FeatureLayer constructor.

### ISSUE THREAD

**Issue by jgravois** [Esri/esri-leaflet#565]
in rc.4 and prior you could instantiate a secure FeatureLayer like this:

```js
L.esri.featureLayer('http://services.arcgis.com/xyz/arcgis/rest/services/shh_secret/FeatureServer/0', {
  token: '<valid token>'
}).addTo(map);
```
starting in rc.5 (when we made a breaking change to the constructors for `Tasks` and `Services`), this no longer works.  i'm noticing duplicated sets of network traffic in my FeatureLayer queries.  the first, includes the appropriate `token` parameter, the second does not.

i dug through the code in `FeatureManager`, `Query`, `Task` and `Request` to try and make sense of what's going on but i haven't been able to figure it out yet.

you can find a live sample [here](https://dl.dropboxusercontent.com/u/59331579/js/feature-layer-token.html).  it will display the relevant behavior until tomorrow at 5:30pm PST (when the token expires).

Esri/esri-leaflet-renderers/issues/59

**Comment by patrickarlt** [comment]
Here is an ultra-minimal JSBin that reproduces this. http://jsbin.com/tehexe/edit?html,console. I've narrowed it down to requesting `f=geojson` with a token.

When I request with `f=geojson` I get a 302 response which redirects me to a URL **without** the token which responds with `Bad Request` rather then a JSON error.

![screen_shot_2015-06-22_at_9_30_24_pm](https://cloud.githubusercontent.com/assets/378557/8299006/58179eb2-1926-11e5-963d-8a6b6c0b1cd0.png)

---

![screen_shot_2015-06-22_at_9_29_58_pm](https://cloud.githubusercontent.com/assets/378557/8299017/83c8a60a-1926-11e5-91da-2e54016bfdc4.png)

![js_bin_-_js_bin](https://cloud.githubusercontent.com/assets/378557/8299037/fef879e0-1926-11e5-86e0-c5ac5faf9161.png)

**Comment by patrickarlt** [comment]
I'll connect with the server team on this tomorrow (Tuesday) and see what is up. We might have to revert this :(

Whats weird is that you cannot reproduce this by just putting the URLs in a browser.

**Comment by jgravois** [comment]
good catch @patrickarlt!

**Comment by patrickarlt** [comment]
The ArcGIS Online/Server team confirmed this is a big with `f=geojson` and has a fix on the dev environment pending. I'll keep this issues up to date.

**Comment by patrickarlt** [comment]
I've confirmed that this is fixed and will be released with the ArcGIS Online 3.8 release in early July.
